motor/discovery/k8s/resolver.go

```diff
@@ -144,15 +144,17 @@ func (r *Resolver) Resolve(ctx context.Context, root *asset.Asset, tc *providers
 		return nil, err
 	}
 
+	if tc.IncludesDiscoveryTarget(common.DiscoveryAuto) {
+		log.Info().Msg("discovery option auto is used. This will detect the assets: cluster, cronjobs, pods, statefulsets, deployments, replicasets, daemonsets")
+	}
+
 	// Only discover cluster and nodes if there are no resource filters. For CI/CD do not
 	// discover the cluster asset at all. In that case that would be the admission review resource
 	// for which we only care if we have explicitly enabled discovery for it.
 	var clusterAsset *asset.Asset
 	ownershipDir := k8s.NewEmptyPlatformIdOwnershipDirectory(clusterIdentifier)
-	if tc.IncludesDiscoveryTarget(common.DiscoveryAll) ||
-		tc.IncludesDiscoveryTarget(common.DiscoveryAuto) ||
-		tc.IncludesDiscoveryTarget(DiscoveryClusters) &&
-			len(resourcesFilter) == 0 && root.Category != asset.AssetCategory_CATEGORY_CICD {
+	if tc.IncludesOneOfDiscoveryTarget(common.DiscoveryAll, common.DiscoveryAuto, DiscoveryClusters) &&
+		len(resourcesFilter) == 0 && root.Category != asset.AssetCategory_CATEGORY_CICD {
```

Comment:
I am not sure why we need this, wouldn't it be easier for the operator to say what he wants to scan?

Reply:
If it is the action, then we should adapt https://github.com/mondoohq/actions/blob/main/k8s-manifest/action.yaml#L47-L51 instead

Reply:
As just discussed, we will check whether we can get ride of the additional conditions in a separate PR.

Reply:
This code is there because for CI/CD scans we don't want to discover the cluster as an asset for the case of the admission controller. I guess we could technically move that logic to the operator itself, indeed